### PR TITLE
refactor(General Ledger)

### LINKED ISSUE #1751: refactor: General Ledger
References issue https://github.com/IMA-WorldHealth/bhima-2.X/issues/1737

The General Ledger module should be updated to meet the latest standards in order to reduce technical debt and give clients required features: 

- [ ] Integrate grid save state (https://github.com/IMA-WorldHealth/bhima-2.X/issues/1738)
  - [ ] Save configuration 
  - [ ] Load default configuration
- [ ] "Menu" dropdown for core utilities
  * Column configuration
  * Download view as CSV
  * Download view as PDF 
- [ ] Remove bhRenderer dropdown 
- [ ] Remove 'Print' action favouring the 'Download as PDF'
- [ ] Ensure links to other registries work correctly

### PR DESCRIPTION
**Pull request by lomamech**
Update general ledger module with the latest standards
- Integrate grid save state
- Use menu dropdown for core utilites
- remove bhRenderer dropdown
- remove print action instead dowloads as PDF

closes #1751

### CODE REVIEW

File: client/src/modules/general-ledger/general-ledger-accounts.ctrl.js [hunk 1]
```diff
@@ -147,13 +152,22 @@ function GeneralLedgerAccountsController(GeneralLedger, Session, Notify,
     flatEntityAccess  : true,
     enableColumnMenus : false,
     appScopeProvider  : vm,
-    onRegisterApi     : onRegisterApiFn,
+    onRegisterApi     : onRegisterApi,
   };
 
-  function onRegisterApiFn(gridApi) {
+  function onRegisterApi(gridApi) {
```
Comment (jniles) [inline client/src/modules/general-ledger/general-ledger-accounts.ctrl.js]: Do you need this function?

File: client/src/modules/general-ledger/general-ledger-accounts.html [hunk 2]
```diff
@@ -6,19 +6,45 @@
...
+            <li role="menuitem">
+              <a href data-method="clear-state" ng-click="GeneralLedgerAccountsCtrl.clearGridState()">
+                <i class="fa fa-close"></i> <span translate>FORM.BUTTONS.CLEAR_GRID_CONFIGURATION</span>
+              </a>
+            </li>
+
+            <li role="separator" class="divider"></li>
+            <li role="menuitem">
+              <a href data-method="print" id="print" ng-click="GeneralLedgerAccountsCtrl.openLedgerReport()">
+                <i class="fa fa-file-pdf-o"></i> <span translate>FORM.BUTTONS.PDF_EXPORT</span>
```
Comment (jniles) [inline client/src/modules/general-ledger/general-ledger-accounts.html:38]: On my machine this isn't translated for some reason.  Does it have a key?

**Review by jniles — APPROVED** [review]
LGTM. :+1:

File: client/src/modules/general-ledger/general-ledger-accounts.html (hunk 2 above)
Comment (lomamech) [reply client/src/modules/general-ledger/general-ledger-accounts.html:38]: Sorry the key is in another Pull Request #1771

Comment (jniles) [reply client/src/modules/general-ledger/general-ledger-accounts.html:38]: Ah, okay.  Sorry for the confusion!

**Pull request merged**